test/e2e/node/kubelet.go

```diff
@@ -479,62 +485,266 @@ var _ = SIGDescribe("kubelet", func() {
+			ginkgo.By("Starting the command")
+			n := "-2"
+			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
+			cmd := tk.KubectlCmd("alpha", "node-logs", "--role", "master", "--query", "kubelet", "--boot", n, "-o", "short")
+			result := runKubectlCommand(cmd, "--role --query --boot")
+			logs := journalctlCommand("-u", "kubelet", "-b", n, "")
+			if result != logs {
+				framework.Failf("Failed to receive the correct kubelet logs for the %s boot", n)
+			}
+		})
+
+		/*
+			Test if kubectl node-logs --role master --query kubelet --boot 2
+			returns an error or not!
+		*/
+
+		ginkgo.It("should return the error of running with boot option value over 0", func() {
+			ginkgo.By("Starting the command")
+			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
+			cmd := tk.KubectlCmd("alpha", "node-logs", "--role", "master", "--query", "kubelet", "--boot", "2")
+			_, _, err := framework.StartCmdAndStreamOutput(cmd)
+			if err != nil {
+				framework.Failf("Failed to start kubectl command: %v", err)
+			}
+			err = cmd.Wait()
+			if err == nil {
+				framework.Failf("Command failed: %v. It was expected an error!", cmd)
+			}
+		})
+
+		/*
+			Test if kubectl node-logs -l kuberentes.io/os=linux --Q kubelet
+			returns the kubelet logs or not!
+		*/
+
+		ginkgo.It("should return the logs for the requested query using the selector kubernetes.io/os=linux", func() {
+			ginkgo.By("Starting the command")
+			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
+			cmd := tk.KubectlCmd("alpha", "node-logs", "-l", "kubernetes.io/os=linux", "--query", "kubelet")
+			result := runKubectlCommand(cmd, "--role -l --query")
+			lookForString("kubelet", result)
+		})
+
+		/*
+			Test if kubectl node-logs --role master --Q kubelet --until-time current date and time
+			returns the kubelet logs or not!
+		*/
+
+		ginkgo.It("should return the logs for the requested query until the current date and time", func() {
+			ginkgo.By("Starting the command")
+			const RFC3339 = "2006-01-02T15:04:05Z07:00"
+			start := time.Now()
+			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
+			cmd := tk.KubectlCmd("alpha", "node-logs", "--role", "master", "--query", "kubelet", "-o", "short", "--until-time", start.Format(RFC3339))
+			result := runKubectlCommand(cmd, "--role --query --until-time")
+			logs := journalctlCommand("-u", "kubelet", "-U", "now", "")
```

Comment:
if this test is run on soak test environment, there will be a LOT of logs. The choice of date to be 2006 may not be ideal.

Reply:
This will print the logs until the current time of the test. By default, kubectl alpha node-logs prints the logs of the current boot. The 2006 is the default date to declare the RFC3339 from what I have read. And if you run the command for --since-time or --until-time with an invalid date, the output is:
`kubectl alpha node-logs -Q query --since-time 20
error: parsing time "20" as "2006-01-02T15:04:05Z07:00": cannot parse "20" as "2006"`